### DISCUSSION TITLE
HTTP requests are forced to be converted to HTTPS

### DISCUSSION THREAD

**Discussion by GG-o1** [projectdiscovery/httpx#1101] [Q&A] [answered]
1、`httpx -u http://127.0.0.1:89 -debug-resp`，it's ok
<img width="446" alt="image" src="https://user-images.githubusercontent.com/32947015/228187082-c3729fbe-8bba-4bc2-9434-3f0a5c932f49.png">

2、`httpx -u http://127.0.0.1:89 -debug-resp -proxy http://127.0.0.1:8080`，set burp proxy，request changed.
<img width="620" alt="image" src="https://user-images.githubusercontent.com/32947015/228187671-19de300d-7721-41e8-882d-9592578f0111.png">

is there any parameter can be set to resolve this?

**Comment by jimen0** [comment]
Might share root cause with https://github.com/projectdiscovery/nuclei/issues/3511. It seems `httpx` first attempts HTTPS and then falls back to HTTP besides HTTP protocol being specified.

This is what I see on `proxify` even after specifically setting `http`.

![image](https://user-images.githubusercontent.com/6826244/230459618-4d6ff829-a9bf-4981-8b54-2c6df5aa5024.png)

**Comment by ehsandeep** [comment, marked as answer]
@GG-o1 if you wanted to utilize the predefined schema for probing you can `-nsf` option to disable httpx's default probing behavior.

```
   -nfs, -no-fallback-scheme          probe with protocol scheme specified in input 
```

```console
echo http://0.0.0.0:8000 | httpx -proxy http://0.0.0.0:8080 -nfs

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

		projectdiscovery.io

http://0.0.0.0:8000
```
